pkg/download/downloader_test.go

```diff
@@ -46,17 +52,29 @@ func TestDownload_badStatusCodeFails(t *testing.T) {
 		http.StatusBadRequest,
 		http.StatusUnauthorized,
 	} {
-		_, _, err := download.Download(download.NewURLDownload(fmt.Sprintf("%s/status/%d", srv.URL, code)))
+		_, _, err := download.Download(testctx, download.NewURLDownload(fmt.Sprintf("%s/status/%d", srv.URL, code)))
 		require.NotNil(t, err, "not failed for code:%d", code)
-		require.Contains(t, err.Error(), "unexpected status code", "wrong message for code %d", code)
+		switch code {
+		case http.StatusNotFound:
+			require.Contains(t, err.Error(), "because it does not exist")
```

Comment:
could you also assert for status code returned using require.AreEqual(<expectedHttpStatusCode>, <actualHttpStatusCode>)